test/functional/treesitter/parser_spec.lua

```diff
@@ -186,12 +187,18 @@ void ui_refresh(void)
       { "type", "primitive_type", 8, 2, 8, 6 },
       { "keyword", "for", 9, 2, 9, 5 },
       { "type", "primitive_type", 9, 7, 9, 13 },
+      -- captured multiple times, see https://github.com/tree-sitter/tree-sitter/issues/1591
+      { "funccall", "expression_statement", 11, 4, 11, 34 },
```

Comment:
not sure if I'm sure what is going on here but it does _not_ seem like an improvement over existing behavior. if it is a unfortunate side-effect of actually useful behavior, I'll expect a test to be added for the latter as well.

Reply:
Sry, probably wasn't to smart to add that specific test case. It's a bug in treesitter I actually spent quite a while tracking down, but it only happens with very complex queries. The bug of course doesn't come up when only one node for each capture is emitted. I'll add a better test case.

Reply:
For example this doesn't happen when the `@funccall` pattern is the only one. The issue I linked is the one I opened in the treesitter repo because of my findings.